README.md

```diff
@@ -469,35 +532,68 @@ dictionary AILanguageModelCreateCoreOptions {
   [EnforceRange] unsigned long topK;
   float temperature;
   sequence<DOMString> expectedInputLanguages;
-}
+  sequence<AILanguageModelPromptType> expectedInputTypes;
+};
 
 dictionary AILanguageModelCreateOptions : AILanguageModelCreateCoreOptions {
   AbortSignal signal;
   AICreateMonitorCallback monitor;
 
   DOMString systemPrompt;
-  sequence<AILanguageModelInitialPrompt> initialPrompts;
+  sequence<AILanguageModelInitialPromptLine> initialPrompts;
+};
+
+dictionary AILanguageModelPromptOptions {
+  AbortSignal signal;
+};
+
+dictionary AILanguageModelCloneOptions {
+  AbortSignal signal;
 };
 
-dictionary AILanguageModelInitialPrompt {
+// The argument to the prompt() method and others like it
+
+typedef (AILanguageModelPromptLine or sequence<AILanguageModelPromptLine>) AILanguageModelPromptInput;
+
+// Initial prompt lines
+
+dictionary AILanguageModelInitialPromptLineDict {
```

Comment:
Is it worthwhile to have separate role enums and dictionaries for initial/not lines?
Would simply throwing exceptions for invalid 'system' role use in the impl suffice?

Reply:
I realize you've explored this thoroughly, but I wonder why the idl couldn't be simpler, like:

dictionary AILanguageModelPromptDict {
  required AILanguageModelPromptRole role = "user";
  required AILanguageModelPromptType type = "text";
  required AILanguageModelPromptData data;
}

typedef (DOMString or AILanguageModelPromptDict) AILanguageModelPrompt;
typedef (AILanguageModelPrompt or sequence<AILanguageModelPrompt>) AILanguageModelPromptInput;

Reply:
> Is it worthwhile to have separate role enums and dictionaries for initial/not lines?
> Would simply throwing exceptions for invalid 'system' role use in the impl suffice?

Yeah that's totally valid. If you as an implementer think that's simpler, we can do it.

> I realize you've explored this thoroughly, but I wonder why the idl couldn't be simpler, like:

(Note that required and default values are not compatible.)

So, your version is mainly about unnesting the content, right?

I think it mostly works, it just departs from what's idiomatic with other chat completion APIs. Hmm.

Reply:
Yeah, I imagined unnesting might slightly simplify IDL, impl, and atypical uses specifying more fields.
`create({initialPrompts: [{role: "system", content: {type: "image", data: pixels}}]})`
vs
`create({initialPrompts: [{role: "system", type: "image", data: pixels}]})`

I'm unfamiliar with rationale behind other APIs' idiomatic designs, so I'd readily defer to folks that know more, and maybe even opt for the more familiar pattern.

Reply:
Based on the lack of developer feedback, including after I tweeted, let me update to the flattened, simpler version.

Reply:
(Note that I also removed the initial vs. non-initial distinction. We'll enforce that in the spec level instead of the IDL level. For now I added a comment to the IDL.)